### PR TITLE
Adapt Keras 2 API and remove redundant last BN/Relu

### PR DESCRIPTION
**Pull request by JihongJu**
Adapt [keras 2 API](https://keras.io/) according to [this blog](https://blog.keras.io/introducing-keras-2.html) #38 

Changes:
A. [functional API](https://keras.io/models/model/):

 - Model.fit_generator:
  + samples_per_epoch -> steps_per_epoch
  + nb_epoch -> epochs

 - Model.__init__
  + input -> inputs
  + output -> outputs

B [Convolutional layers](https://keras.io/layers/convolutional/):

 - Convolution2D -> Conv2D
  + nb_filter -> filters
  + nb_row, nb_col -> kernel_size
  + subsample -> strides
  + border_mode -> padding
  + init -> kernel_initializer
  + W_regularizer - > kernel_regularizer

B. BatchNormalization (remove mode)

C. [Polling layers](https://keras.io/layers/pooling/)
 - border_mode -> padding

D. [merge](https://keras.io/layers/merge/)
 -  merge(inputs, mode='sum')-> add(inputs)

### CODE REVIEW

File: cifar10.py [hunk 1]
```diff
@@ -80,7 +80,7 @@
 
     # Fit the model on the batches generated by datagen.flow().
     model.fit_generator(datagen.flow(X_train, Y_train, batch_size=batch_size),
-                        samples_per_epoch=X_train.shape[0],
+                        steps_per_epoch=X_train.shape[0],
```
Comment (raghakot) [inline cifar10.py]: this needs to be X_train.shape[0] // batch_size

Comment (raghakot) [inline cifar10.py]: I can update it though. Thanks for the PR.
